# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsParent.cpp
# Commit: eff806e9615d
# Full Hash: eff806e9615d165bc8cf2adcf43021dc9a5a39c4
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-10 15:51:07
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Unified handling of literal strings using NS_LITERAL_(C)STRING and removed some duplications. r=ttung
#   
#   Harmonized and provided a rationale for defining statement parameter names literally vs. by named constant,
#   and referencing them in binding calls by index vs. by name.
#   
# ==============================================================================

diff -r b9c38ead5e55 -r eff806e9615d dom/indexedDB/ActorsParent.cpp
--- a/dom/indexedDB/ActorsParent.cpp	Tue Sep 10 16:48:00 2019 +0300
+++ b/dom/indexedDB/ActorsParent.cpp	Tue Sep 10 16:48:13 2019 +0300
@@ -246,23 +246,60 @@
 
 const uint32_t kFileCopyBufferSize = 32768;
 
-#define JOURNAL_DIRECTORY_NAME "journals"
-
-const char kFileManagerDirectoryNameSuffix[] = ".files";
-const char kSQLiteSuffix[] = ".sqlite";
-const char kSQLiteJournalSuffix[] = ".sqlite-journal";
-const char kSQLiteSHMSuffix[] = ".sqlite-shm";
-const char kSQLiteWALSuffix[] = ".sqlite-wal";
+constexpr auto kJournalDirectoryName = NS_LITERAL_STRING("journals");
+
+constexpr auto kFileManagerDirectoryNameSuffix = NS_LITERAL_STRING(".files");
+constexpr auto kSQLiteSuffix = NS_LITERAL_STRING(".sqlite");
+constexpr auto kSQLiteJournalSuffix = NS_LITERAL_STRING(".sqlite-journal");
+constexpr auto kSQLiteSHMSuffix = NS_LITERAL_STRING(".sqlite-shm");
+constexpr auto kSQLiteWALSuffix = NS_LITERAL_STRING(".sqlite-wal");
 
 const char kPrefIndexedDBEnabled[] = "dom.indexedDB.enabled";
 
 const char kPrefFileHandleEnabled[] = "dom.fileHandle.enabled";
 
-#define IDB_PREFIX "indexedDB"
-
-#define PERMISSION_STRING_CHROME_BASE IDB_PREFIX "-chrome-"
-#define PERMISSION_STRING_CHROME_READ_SUFFIX "-read"
-#define PERMISSION_STRING_CHROME_WRITE_SUFFIX "-write"
+constexpr auto kPermissionStringBase = NS_LITERAL_CSTRING("indexedDB-chrome-");
+constexpr auto kPermissionReadSuffix = NS_LITERAL_CSTRING("-read");
+constexpr auto kPermissionWriteSuffix = NS_LITERAL_CSTRING("-write");
+
+// The following constants define all names of binding parameters in statements,
+// where they are bound by name. This should include all parameter names which
+// are bound by name. Binding may be done by index when the statement definition
+// and binding are done in the same local scope, and no other reasons prevent
+// using the indexes (e.g. multiple statement variants with differing number or
+// order of parameters). Neither the styles of specifying parameter names
+// (literally vs. via these constants) nor the binding styles (by index vs. by
+// name) should not be mixed for the same statement. The decision must be made
+// for each statement based on the proximity of statement and binding calls.
+constexpr auto kStmtParamNameCurrentKey = NS_LITERAL_CSTRING("current_key");
+constexpr auto kStmtParamNameRangeKey = NS_LITERAL_CSTRING("range_key");
+constexpr auto kStmtParamNameObjectKey = NS_LITERAL_CSTRING("object_key");
+constexpr auto kStmtParamNameLowerKey = NS_LITERAL_CSTRING("lower_key");
+constexpr auto kStmtParamNameUpperKey = NS_LITERAL_CSTRING("upper_key");
+constexpr auto kStmtParamNameKey = NS_LITERAL_CSTRING("key");
+constexpr auto kStmtParamNameObjectStoreId =
+    NS_LITERAL_CSTRING("object_store_id");
+constexpr auto kStmtParamNameIndexId = NS_LITERAL_CSTRING("index_id");
+// TODO: Maybe the uses of kStmtParamNameId should be replaced by more
+// specific constants such as kStmtParamNameObjectStoreId.
+constexpr auto kStmtParamNameId = NS_LITERAL_CSTRING("id");
+constexpr auto kStmtParamNameValue = NS_LITERAL_CSTRING("value");
+constexpr auto kStmtParamNameObjectDataKey =
+    NS_LITERAL_CSTRING("object_data_key");
+constexpr auto kStmtParamNameIndexDataValues =
+    NS_LITERAL_CSTRING("index_data_values");
+constexpr auto kStmtParamNameData = NS_LITERAL_CSTRING("data");
+constexpr auto kStmtParamNameFileIds = NS_LITERAL_CSTRING("file_ids");
+constexpr auto kStmtParamNameValueLocale = NS_LITERAL_CSTRING("value_locale");
+
+// The following constants define some names of columns in tables, which are
+// referred to in remote locations, e.g. in calls to
+// GetBindingClauseForKeyRange.
+constexpr auto kColumnNameKey = NS_LITERAL_CSTRING("key");
+constexpr auto kColumnNameValue = NS_LITERAL_CSTRING("value");
+
+// SQL fragments used at multiple locations.
+constexpr auto kOpenLimit = NS_LITERAL_CSTRING(" LIMIT ");
 
 // The deletion marker file is created before RemoveDatabaseFilesAndDirectory
 // begins deleting a database. It is removed as the last step of deletion. If a
@@ -271,7 +308,8 @@
 // are removed. The primary goal of this mechanism is to avoid situations where
 // a database has been partially deleted, leading to inconsistent state for the
 // origin.
-#define IDB_DELETION_MARKER_FILE_PREFIX "idb-deleting-"
+constexpr auto kIdbDeletionMarkerFilePrefix =
+    NS_LITERAL_STRING("idb-deleting-");
 
 const uint32_t kDeleteTimeoutMs = 1000;
 
@@ -300,14 +338,6 @@
 
 #endif
 
-template <size_t N>
-constexpr size_t LiteralStringLength(const char (&aArr)[N]) {
-  static_assert(N, "Zero-length string literal?!");
-
-  // Don't include the null terminator.
-  return N - 1;
-}
-
 /*******************************************************************************
  * Metadata classes
  ******************************************************************************/
@@ -1213,6 +1243,8 @@
     return rv;
   }
 
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   rv = aConnection->CreateStatement(
       NS_LITERAL_CSTRING("INSERT INTO database (name, version, dataVersion) "
                          "VALUES (:name, :version, :dataVersion)"),
@@ -3203,6 +3235,9 @@
   // Update the |database| table to include the origin, vacuum information, and
   // apply the WITHOUT ROWID optimization.
   nsCOMPtr<mozIStorageStatement> stmt;
+
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   rv = aConnection->CreateStatement(
       NS_LITERAL_CSTRING("INSERT INTO database_upgrade "
                          "SELECT name, :origin, version, 0, 0, 0 "
@@ -3212,7 +3247,7 @@
     return rv;
   }
 
-  rv = stmt->BindUTF8StringByName(NS_LITERAL_CSTRING("origin"), aOrigin);
+  rv = stmt->BindUTF8StringByIndex(0, aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -3720,6 +3755,8 @@
   AUTO_PROFILER_LABEL("UpgradeSchemaFrom22_0To23_0", DOM);
 
   nsCOMPtr<mozIStorageStatement> stmt;
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   nsresult rv =
       aConnection->CreateStatement(NS_LITERAL_CSTRING("UPDATE database "
                                                       "SET origin = :origin;"),
@@ -3728,7 +3765,7 @@
     return rv;
   }
 
-  rv = stmt->BindUTF8StringByName(NS_LITERAL_CSTRING("origin"), aOrigin);
+  rv = stmt->BindUTF8StringByIndex(0, aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -3902,7 +3939,7 @@
   if (aTelemetryId) {
     telemetryFilenameClause.AssignLiteral("&telemetryFilename=indexedDB-");
     telemetryFilenameClause.AppendInt(aTelemetryId);
-    telemetryFilenameClause.AppendLiteral(".sqlite");
+    telemetryFilenameClause.Append(NS_ConvertUTF16toUTF8(kSQLiteSuffix));
   }
 
   rv = NS_MutateURI(mutator)
@@ -3925,7 +3962,7 @@
   MOZ_ASSERT(!NS_IsMainThread());
   MOZ_ASSERT(aConnection);
 
-  static const char kBuiltInPragmas[] =
+  static constexpr auto kBuiltInPragmas = NS_LITERAL_CSTRING(
       // We use foreign keys in DEBUG builds only because there is a performance
       // cost to using them.
       "PRAGMA foreign_keys = "
@@ -3945,10 +3982,9 @@
 
       // We aggressively truncate the database file when idle so don't bother
       // overwriting the WAL with 0 during active periods.
-      "PRAGMA secure_delete = OFF;";
-
-  nsresult rv = aConnection->ExecuteSimpleSQL(nsDependentCString(
-      kBuiltInPragmas, LiteralStringLength(kBuiltInPragmas)));
+      "PRAGMA secure_delete = OFF;");
+
+  nsresult rv = aConnection->ExecuteSimpleSQL(kBuiltInPragmas);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -4299,6 +4335,8 @@
       MOZ_ASSERT(schemaVersion == kSQLiteSchemaVersion);
 
       nsCOMPtr<mozIStorageStatement> stmt;
+      // The parameter names are not used, parameters are bound by index only
+      // locally in the same function.
       nsresult rv = connection->CreateStatement(
           NS_LITERAL_CSTRING("INSERT INTO database (name, origin) "
                              "VALUES (:name, :origin)"),
@@ -4307,12 +4345,12 @@
         return rv;
       }
 
-      rv = stmt->BindStringByName(NS_LITERAL_CSTRING("name"), aName);
-      if (NS_WARN_IF(NS_FAILED(rv))) {
-        return rv;
-      }
-
-      rv = stmt->BindUTF8StringByName(NS_LITERAL_CSTRING("origin"), aOrigin);
+      rv = stmt->BindStringByIndex(0, aName);
+      if (NS_WARN_IF(NS_FAILED(rv))) {
+        return rv;
+      }
+
+      rv = stmt->BindUTF8StringByIndex(1, aOrigin);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -4519,6 +4557,8 @@
       MOZ_ASSERT(vacuumTime);
 
       nsCOMPtr<mozIStorageStatement> vacuumTimeStmt;
+      // The parameter names are not used, parameters are bound by index only
+      // locally in the same function.
       rv = connection->CreateStatement(
           NS_LITERAL_CSTRING("UPDATE database "
                              "SET last_vacuum_time = :time"
@@ -4528,14 +4568,12 @@
         return rv;
       }
 
-      rv = vacuumTimeStmt->BindInt64ByName(NS_LITERAL_CSTRING("time"),
-                                           vacuumTime);
-      if (NS_WARN_IF(NS_FAILED(rv))) {
-        return rv;
-      }
-
-      rv =
-          vacuumTimeStmt->BindInt64ByName(NS_LITERAL_CSTRING("size"), fileSize);
+      rv = vacuumTimeStmt->BindInt64ByIndex(0, vacuumTime);
+      if (NS_WARN_IF(NS_FAILED(rv))) {
+        return rv;
+      }
+
+      rv = vacuumTimeStmt->BindInt64ByIndex(1, fileSize);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -4634,7 +4672,7 @@
   MOZ_ASSERT(!NS_IsMainThread());
   MOZ_ASSERT(!IsOnBackgroundThread());
   MOZ_ASSERT(!aDatabaseFilePath.IsEmpty());
-  MOZ_ASSERT(StringEndsWith(aDatabaseFilePath, NS_LITERAL_STRING(".sqlite")));
+  MOZ_ASSERT(StringEndsWith(aDatabaseFilePath, kSQLiteSuffix));
   MOZ_ASSERT(aConnection);
 
   nsCOMPtr<nsIFile> dbFile = GetFileForPath(aDatabaseFilePath);
@@ -5407,8 +5445,9 @@
                                           const nsCString& aLocale);
 
   static void AppendConditionClause(const nsACString& aColumnName,
-                                    const nsACString& aArgName, bool aLessThan,
-                                    bool aEquals, nsAutoCString& aResult);
+                                    const nsACString& aStatementParameterName,
+                                    bool aLessThan, bool aEquals,
+                                    nsAutoCString& aResult);
 
   static nsresult GetUniqueIndexTableForObjectStore(
       TransactionBase* aTransaction, int64_t aObjectStoreId,
@@ -8800,8 +8839,7 @@
     return rv;
   }
 
-  rv = markerFile->Append(NS_LITERAL_STRING(IDB_DELETION_MARKER_FILE_PREFIX) +
-                          aDatabaseNameBase);
+  rv = markerFile->Append(kIdbDeletionMarkerFilePrefix + aDatabaseNameBase);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -8862,46 +8900,31 @@
   }
 
   // The database file counts towards quota.
-  nsAutoString filename = aFilenameBase + NS_LITERAL_STRING(".sqlite");
-
-  rv = DeleteFile(aBaseDirectory, filename, aQuotaManager, aPersistenceType,
-                  aGroup, aOrigin);
+  rv = DeleteFile(aBaseDirectory, aFilenameBase + kSQLiteSuffix, aQuotaManager,
+                  aPersistenceType, aGroup, aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   // .sqlite-journal files don't count towards quota.
-  const NS_ConvertASCIItoUTF16 journalSuffix(
-      kSQLiteJournalSuffix, LiteralStringLength(kSQLiteJournalSuffix));
-
-  filename = aFilenameBase + journalSuffix;
-
-  rv = DeleteFile(aBaseDirectory, filename, /* doesn't count */ nullptr,
-                  aPersistenceType, aGroup, aOrigin);
+  rv = DeleteFile(aBaseDirectory, aFilenameBase + kSQLiteJournalSuffix,
+                  /* doesn't count */ nullptr, aPersistenceType, aGroup,
+                  aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   // .sqlite-shm files don't count towards quota.
-  const NS_ConvertASCIItoUTF16 shmSuffix(kSQLiteSHMSuffix,
-                                         LiteralStringLength(kSQLiteSHMSuffix));
-
-  filename = aFilenameBase + shmSuffix;
-
-  rv = DeleteFile(aBaseDirectory, filename, /* doesn't count */ nullptr,
-                  aPersistenceType, aGroup, aOrigin);
+  rv = DeleteFile(aBaseDirectory, aFilenameBase + kSQLiteSHMSuffix,
+                  /* doesn't count */ nullptr, aPersistenceType, aGroup,
+                  aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   // .sqlite-wal files do count towards quota.
-  const NS_ConvertASCIItoUTF16 walSuffix(kSQLiteWALSuffix,
-                                         LiteralStringLength(kSQLiteWALSuffix));
-
-  filename = aFilenameBase + walSuffix;
-
-  rv = DeleteFile(aBaseDirectory, filename, aQuotaManager, aPersistenceType,
-                  aGroup, aOrigin);
+  rv = DeleteFile(aBaseDirectory, aFilenameBase + kSQLiteWALSuffix,
+                  aQuotaManager, aPersistenceType, aGroup, aOrigin);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -8913,11 +8936,7 @@
   }
 
   // The files directory counts towards quota.
-  const NS_ConvertASCIItoUTF16 filesSuffix(
-      kFileManagerDirectoryNameSuffix,
-      LiteralStringLength(kFileManagerDirectoryNameSuffix));
-
-  rv = fmDirectory->Append(aFilenameBase + filesSuffix);
+  rv = fmDirectory->Append(aFilenameBase + kFileManagerDirectoryNameSuffix);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -9129,11 +9148,9 @@
   MOZ_ALWAYS_SUCCEEDS(aFile->GetLeafName(filename));
 
   // Make sure we were given a database file.
-  NS_NAMED_LITERAL_STRING(sqliteExtension, ".sqlite");
-
-  MOZ_ASSERT(StringEndsWith(filename, sqliteExtension));
-
-  filename.Truncate(filename.Length() - sqliteExtension.Length());
+  MOZ_ASSERT(StringEndsWith(filename, kSQLiteSuffix));
+
+  filename.Truncate(filename.Length() - kSQLiteSuffix.Length());
 
   // Get the "idb" directory.
   nsCOMPtr<nsIFile> idbDirectory;
@@ -10587,6 +10604,8 @@
 
   nsresult rv;
   if (!mUpdateStatement) {
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     rv = connection->GetCachedStatement(
         NS_LITERAL_CSTRING("UPDATE file "
                            "SET refcount = refcount + :delta "
@@ -10599,12 +10618,12 @@
 
   mozStorageStatementScoper updateScoper(mUpdateStatement);
 
-  rv = mUpdateStatement->BindInt32ByName(NS_LITERAL_CSTRING("delta"), aDelta);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = mUpdateStatement->BindInt64ByName(NS_LITERAL_CSTRING("id"), aId);
+  rv = mUpdateStatement->BindInt32ByIndex(0, aDelta);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = mUpdateStatement->BindInt64ByIndex(1, aId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -10622,6 +10641,8 @@
 
   if (rows > 0) {
     if (!mSelectStatement) {
+      // The parameter names are not used, parameters are bound by index only
+      // locally in the same function.
       rv = connection->GetCachedStatement(NS_LITERAL_CSTRING("SELECT id "
                                                              "FROM file "
                                                              "WHERE id = :id"),
@@ -10633,7 +10654,7 @@
 
     mozStorageStatementScoper selectScoper(mSelectStatement);
 
-    rv = mSelectStatement->BindInt64ByName(NS_LITERAL_CSTRING("id"), aId);
+    rv = mSelectStatement->BindInt64ByIndex(0, aId);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -10654,6 +10675,8 @@
   }
 
   if (!mInsertStatement) {
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     rv = connection->GetCachedStatement(
         NS_LITERAL_CSTRING("INSERT INTO file (id, refcount) "
                            "VALUES(:id, :delta)"),
@@ -10665,12 +10688,12 @@
 
   mozStorageStatementScoper insertScoper(mInsertStatement);
 
-  rv = mInsertStatement->BindInt64ByName(NS_LITERAL_CSTRING("id"), aId);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = mInsertStatement->BindInt32ByName(NS_LITERAL_CSTRING("delta"), aDelta);
+  rv = mInsertStatement->BindInt64ByIndex(0, aId);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = mInsertStatement->BindInt32ByIndex(1, aDelta);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -15248,7 +15271,7 @@
     return rv;
   }
 
-  rv = journalDirectory->Append(NS_LITERAL_STRING(JOURNAL_DIRECTORY_NAME));
+  rv = journalDirectory->Append(kJournalDirectoryName);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -15511,7 +15534,7 @@
     return rv;
   }
 
-  rv = journalDirectory->Append(NS_LITERAL_STRING(JOURNAL_DIRECTORY_NAME));
+  rv = journalDirectory->Append(kJournalDirectoryName);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -15561,6 +15584,8 @@
       }
 
       nsCOMPtr<mozIStorageStatement> stmt;
+      // The parameter names are not used, parameters are bound by index only
+      // locally in the same function.
       rv = connection->CreateStatement(
           NS_LITERAL_CSTRING("SELECT name, (name IN (SELECT id FROM file)) "
                              "FROM fs "
@@ -15576,7 +15601,7 @@
         return rv;
       }
 
-      rv = stmt->BindStringByName(NS_LITERAL_CSTRING("path"), path);
+      rv = stmt->BindStringByIndex(0, path);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -15676,7 +15701,7 @@
       return rv;
     }
 
-    if (leafName.EqualsLiteral(JOURNAL_DIRECTORY_NAME)) {
+    if (leafName.Equals(kJournalDirectoryName)) {
       continue;
     }
 
@@ -15842,17 +15867,14 @@
     return rv;
   }
 
-  const NS_ConvertASCIItoUTF16 filesSuffix(
-      kFileManagerDirectoryNameSuffix,
-      LiteralStringLength(kFileManagerDirectoryNameSuffix));
-
   for (uint32_t count = subdirsToProcess.Length(), i = 0; i < count; i++) {
     const nsString& subdirName = subdirsToProcess[i];
 
     // If the directory has the correct suffix then it should exist in
     // databaseFilenames.
     nsDependentSubstring subdirNameBase;
-    if (GetBaseFilename(subdirName, filesSuffix, subdirNameBase)) {
+    if (GetBaseFilename(subdirName, kFileManagerDirectoryNameSuffix,
+                        subdirNameBase)) {
       Unused << NS_WARN_IF(!databaseFilenames.GetEntry(subdirNameBase));
 
       continue;
@@ -15862,7 +15884,7 @@
     // it. Check to see if we have a database that references this directory.
     nsString subdirNameWithSuffix;
     if (databaseFilenames.GetEntry(subdirName)) {
-      subdirNameWithSuffix = subdirName + filesSuffix;
+      subdirNameWithSuffix = subdirName + kFileManagerDirectoryNameSuffix;
     } else {
       // Windows doesn't allow a directory to end with a dot ('.'), so we have
       // to check that possibility here too.
@@ -15872,7 +15894,8 @@
       if (NS_WARN_IF(!databaseFilenames.GetEntry(subdirNameWithDot))) {
         continue;
       }
-      subdirNameWithSuffix = subdirNameWithDot + filesSuffix;
+      subdirNameWithSuffix =
+          subdirNameWithDot + kFileManagerDirectoryNameSuffix;
     }
 
     // We do have a database that uses this directory so we should rename it
@@ -16000,16 +16023,13 @@
     return rv;
   }
 
-  const NS_ConvertASCIItoUTF16 filesSuffix(
-      kFileManagerDirectoryNameSuffix,
-      LiteralStringLength(kFileManagerDirectoryNameSuffix));
-
   for (uint32_t count = subdirsToProcess.Length(), i = 0; i < count; i++) {
     const nsString& subdirName = subdirsToProcess[i];
 
     // The directory must have the correct suffix.
     nsDependentSubstring subdirNameBase;
-    if (NS_WARN_IF(!GetBaseFilename(subdirName, filesSuffix, subdirNameBase))) {
+    if (NS_WARN_IF(!GetBaseFilename(subdirName, kFileManagerDirectoryNameSuffix,
+                                    subdirNameBase))) {
       // If there is an unexpected directory in the idb directory, trying to
       // delete at first instead of breaking the whole initialization.
       if (NS_WARN_IF(NS_FAILED(DeleteFilesNoQuota(directory, subdirName)))) {
@@ -16046,10 +16066,7 @@
     }
   }
 
-  const NS_ConvertASCIItoUTF16 sqliteSuffix(kSQLiteSuffix,
-                                            LiteralStringLength(kSQLiteSuffix));
-  const NS_ConvertASCIItoUTF16 walSuffix(kSQLiteWALSuffix,
-                                         LiteralStringLength(kSQLiteWALSuffix));
+  const auto sqliteSuffix = kSQLiteSuffix;
 
   for (auto iter = databaseFilenames.ConstIter(); !iter.Done() && !aCanceled;
        iter.Next()) {
@@ -16062,7 +16079,8 @@
       return rv;
     }
 
-    rv = fmDirectory->Append(databaseFilename + filesSuffix);
+    rv =
+        fmDirectory->Append(databaseFilename + kFileManagerDirectoryNameSuffix);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       REPORT_TELEMETRY_INIT_ERR(kQuotaExternalError, IDB_Append);
       return rv;
@@ -16089,7 +16107,7 @@
         return rv;
       }
 
-      rv = walFile->Append(databaseFilename + walSuffix);
+      rv = walFile->Append(databaseFilename + kSQLiteWALSuffix);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         REPORT_TELEMETRY_INIT_ERR(kQuotaExternalError, IDB_Append3);
         return rv;
@@ -16441,14 +16459,7 @@
     return rv;
   }
 
-  const NS_ConvertASCIItoUTF16 sqliteSuffix(kSQLiteSuffix,
-                                            LiteralStringLength(kSQLiteSuffix));
-  const NS_ConvertASCIItoUTF16 journalSuffix(
-      kSQLiteJournalSuffix, LiteralStringLength(kSQLiteJournalSuffix));
-  const NS_ConvertASCIItoUTF16 shmSuffix(kSQLiteSHMSuffix,
-                                         LiteralStringLength(kSQLiteSHMSuffix));
-  const NS_ConvertASCIItoUTF16 walSuffix(kSQLiteWALSuffix,
-                                         LiteralStringLength(kSQLiteWALSuffix));
+  const auto sqliteSuffix = kSQLiteSuffix;
 
   nsCOMPtr<nsIFile> file;
   while (NS_SUCCEEDED((rv = entries->GetNextFile(getter_AddRefs(file)))) &&
@@ -16471,11 +16482,9 @@
     }
 
     if (aObsoleteFilenames &&
-        StringBeginsWith(leafName,
-                         NS_LITERAL_STRING(IDB_DELETION_MARKER_FILE_PREFIX))) {
-      const uint32_t start = 13;
-      MOZ_ASSERT(start == LiteralStringLength(IDB_DELETION_MARKER_FILE_PREFIX));
-      aObsoleteFilenames->PutEntry(Substring(leafName, start));
+        StringBeginsWith(leafName, kIdbDeletionMarkerFilePrefix)) {
+      aObsoleteFilenames->PutEntry(
+          Substring(leafName, kIdbDeletionMarkerFilePrefix.Length()));
       continue;
     }
 
@@ -16494,14 +16503,14 @@
     // Skip SQLite temporary files. These files take up space on disk but will
     // be deleted as soon as the database is opened, so we don't count them
     // towards quota.
-    if (StringEndsWith(leafName, journalSuffix) ||
-        StringEndsWith(leafName, shmSuffix)) {
+    if (StringEndsWith(leafName, kSQLiteJournalSuffix) ||
+        StringEndsWith(leafName, kSQLiteSHMSuffix)) {
       continue;
     }
 
     // The SQLite WAL file does count towards quota, but it is handled below
     // once we find the actual database file.
-    if (StringEndsWith(leafName, walSuffix)) {
+    if (StringEndsWith(leafName, kSQLiteWALSuffix)) {
       continue;
     }
 
@@ -16549,11 +16558,6 @@
     return NS_OK;
   }
 
-  const NS_ConvertASCIItoUTF16 journalSuffix(
-      kSQLiteJournalSuffix, LiteralStringLength(kSQLiteJournalSuffix));
-  const NS_ConvertASCIItoUTF16 shmSuffix(kSQLiteSHMSuffix,
-                                         LiteralStringLength(kSQLiteSHMSuffix));
-
   nsCOMPtr<nsIFile> file;
   while (NS_SUCCEEDED((rv = entries->GetNextFile(getter_AddRefs(file)))) &&
          file && !aCanceled) {
@@ -16566,8 +16570,7 @@
     // Ignore the markerfiles. Note: the unremoved files can still be calculated
     // here as long as its size matches the size recorded in QuotaManager
     // in-memory objects.
-    if (StringBeginsWith(leafName,
-                         NS_LITERAL_STRING(IDB_DELETION_MARKER_FILE_PREFIX))) {
+    if (StringBeginsWith(leafName, kIdbDeletionMarkerFilePrefix)) {
       continue;
     }
 
@@ -16577,8 +16580,8 @@
     }
 
     // Journal files and sqlite-shm files don't count towards usage.
-    if (StringEndsWith(leafName, journalSuffix) ||
-        StringEndsWith(leafName, shmSuffix)) {
+    if (StringEndsWith(leafName, kSQLiteJournalSuffix) ||
+        StringEndsWith(leafName, kSQLiteSHMSuffix)) {
       continue;
     }
 
@@ -16606,7 +16609,7 @@
           return rv;
         }
 
-        if (!leafName.EqualsLiteral(JOURNAL_DIRECTORY_NAME)) {
+        if (!leafName.Equals(kJournalDirectoryName)) {
           NS_WARNING("Unknown directory found!");
         }
       }
@@ -17003,7 +17006,6 @@
                 "Something changed with available persistence types!");
 
   NS_NAMED_LITERAL_STRING(idbDirName, IDB_DIRECTORY_NAME);
-  NS_NAMED_LITERAL_STRING(sqliteExtension, ".sqlite");
 
   for (const PersistenceType persistenceType : kPersistenceTypes) {
     // Loop over "<persistence>" directories.
@@ -17160,7 +17162,7 @@
           return rv;
         }
 
-        if (!StringEndsWith(idbFilePath, sqliteExtension)) {
+        if (!StringEndsWith(idbFilePath, kSQLiteSuffix)) {
           continue;
         }
 
@@ -17862,21 +17864,23 @@
   MOZ_ASSERT(fileSize > 0);
 
   nsCOMPtr<mozIStorageStatement> stmt;
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   rv = aConnection->CreateStatement(
       NS_LITERAL_CSTRING("UPDATE database "
                          "SET last_vacuum_time = :time"
                          ", last_vacuum_size = :size;"),
-      getter_AddRefs(stmt));
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("time"), vacuumTime);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("size"), fileSize);
+                               getter_AddRefs(stmt));
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return;
+  }
+
+  rv = stmt->BindInt64ByIndex(0, vacuumTime);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return;
+  }
+
+  rv = stmt->BindInt64ByIndex(1, fileSize);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return;
   }
@@ -18062,12 +18066,11 @@
 
   NS_NAMED_LITERAL_CSTRING(andStr, " AND ");
   NS_NAMED_LITERAL_CSTRING(spacecolon, " :");
-  NS_NAMED_LITERAL_CSTRING(lowerKey, "lower_key");
 
   if (aKeyRange.isOnly()) {
     // Both keys equal.
     aBindingClause = andStr + aKeyColumnName + NS_LITERAL_CSTRING(" =") +
-                     spacecolon + lowerKey;
+                     spacecolon + kStmtParamNameLowerKey;
     return;
   }
 
@@ -18080,7 +18083,7 @@
     if (!aKeyRange.lowerOpen()) {
       aBindingClause.AppendLiteral("=");
     }
-    aBindingClause.Append(spacecolon + lowerKey);
+    aBindingClause.Append(spacecolon + kStmtParamNameLowerKey);
   }
 
   if (!aKeyRange.upper().IsUnset()) {
@@ -18090,7 +18093,7 @@
     if (!aKeyRange.upperOpen()) {
       aBindingClause.AppendLiteral("=");
     }
-    aBindingClause.Append(spacecolon + NS_LITERAL_CSTRING("upper_key"));
+    aBindingClause.Append(spacecolon + kStmtParamNameUpperKey);
   }
 
   MOZ_ASSERT(!aBindingClause.IsEmpty());
@@ -18296,8 +18299,7 @@
   nsresult rv = NS_OK;
 
   if (!aKeyRange.lower().IsUnset()) {
-    rv = aKeyRange.lower().BindToStatement(aStatement,
-                                           NS_LITERAL_CSTRING("lower_key"));
+    rv = aKeyRange.lower().BindToStatement(aStatement, kStmtParamNameLowerKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18308,8 +18310,7 @@
   }
 
   if (!aKeyRange.upper().IsUnset()) {
-    rv = aKeyRange.upper().BindToStatement(aStatement,
-                                           NS_LITERAL_CSTRING("upper_key"));
+    rv = aKeyRange.upper().BindToStatement(aStatement, kStmtParamNameUpperKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18339,7 +18340,7 @@
                  : NS_ERROR_DOM_INDEXEDDB_DATA_ERR;
     }
 
-    rv = lower.BindToStatement(aStatement, NS_LITERAL_CSTRING("lower_key"));
+    rv = lower.BindToStatement(aStatement, kStmtParamNameLowerKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18360,7 +18361,7 @@
                  : NS_ERROR_DOM_INDEXEDDB_DATA_ERR;
     }
 
-    rv = upper.BindToStatement(aStatement, NS_LITERAL_CSTRING("upper_key"));
+    rv = upper.BindToStatement(aStatement, kStmtParamNameUpperKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18370,10 +18371,9 @@
 }
 
 // static
-void DatabaseOperationBase::AppendConditionClause(const nsACString& aColumnName,
-                                                  const nsACString& aArgName,
-                                                  bool aLessThan, bool aEquals,
-                                                  nsAutoCString& aResult) {
+void DatabaseOperationBase::AppendConditionClause(
+    const nsACString& aColumnName, const nsACString& aStatementParameterName,
+    bool aLessThan, bool aEquals, nsAutoCString& aResult) {
   aResult +=
       NS_LITERAL_CSTRING(" AND ") + aColumnName + NS_LITERAL_CSTRING(" ");
 
@@ -18387,7 +18387,7 @@
     aResult.Append('=');
   }
 
-  aResult += NS_LITERAL_CSTRING(" :") + aArgName;
+  aResult += NS_LITERAL_CSTRING(" :") + aStatementParameterName;
 }
 
 // static
@@ -18495,12 +18495,6 @@
     return NS_OK;
   }
 
-  NS_NAMED_LITERAL_CSTRING(objectStoreIdString, "object_store_id");
-  NS_NAMED_LITERAL_CSTRING(objectDataKeyString, "object_data_key");
-  NS_NAMED_LITERAL_CSTRING(indexIdString, "index_id");
-  NS_NAMED_LITERAL_CSTRING(valueString, "value");
-  NS_NAMED_LITERAL_CSTRING(valueLocaleString, "value_locale");
-
   DatabaseConnection::CachedStatement insertUniqueStmt;
   DatabaseConnection::CachedStatement insertStmt;
 
@@ -18519,8 +18513,12 @@
           NS_LITERAL_CSTRING("INSERT INTO unique_index_data "
                              "(index_id, value, object_store_id, "
                              "object_data_key, value_locale) "
-                             "VALUES (:index_id, :value, :object_store_id, "
-                             ":object_data_key, :value_locale);"),
+                             "VALUES (:") +
+              kStmtParamNameIndexId + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameValue + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameObjectDataKey + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameValueLocale + NS_LITERAL_CSTRING(");"),
           &stmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
@@ -18530,35 +18528,39 @@
           NS_LITERAL_CSTRING("INSERT OR IGNORE INTO index_data "
                              "(index_id, value, object_data_key, "
                              "object_store_id, value_locale) "
-                             "VALUES (:index_id, :value, :object_data_key, "
-                             ":object_store_id, :value_locale);"),
+                             "VALUES (:") +
+              kStmtParamNameIndexId + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameValue + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameObjectDataKey + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(", :") +
+              kStmtParamNameValueLocale + NS_LITERAL_CSTRING(");"),
           &stmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
     }
 
-    rv = stmt->BindInt64ByName(indexIdString, info.mIndexId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = info.mKey.BindToStatement(stmt, valueString);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = info.mSortKey.BindToStatement(stmt, valueLocaleString);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = stmt->BindInt64ByName(objectStoreIdString, aObjectStoreId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = aObjectStoreKey.BindToStatement(stmt, objectDataKeyString);
+    rv = stmt->BindInt64ByName(kStmtParamNameIndexId, info.mIndexId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = info.mKey.BindToStatement(stmt, kStmtParamNameValue);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = info.mSortKey.BindToStatement(stmt, kStmtParamNameValueLocale);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, aObjectStoreId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = aObjectStoreKey.BindToStatement(stmt, kStmtParamNameObjectDataKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18603,10 +18605,6 @@
     return NS_OK;
   }
 
-  NS_NAMED_LITERAL_CSTRING(indexIdString, "index_id");
-  NS_NAMED_LITERAL_CSTRING(valueString, "value");
-  NS_NAMED_LITERAL_CSTRING(objectDataKeyString, "object_data_key");
-
   DatabaseConnection::CachedStatement deleteUniqueStmt;
   DatabaseConnection::CachedStatement deleteStmt;
 
@@ -18623,8 +18621,9 @@
     } else if (indexValue.mUnique) {
       rv = aConnection->GetCachedStatement(
           NS_LITERAL_CSTRING("DELETE FROM unique_index_data "
-                             "WHERE index_id = :index_id "
-                             "AND value = :value;"),
+                             "WHERE index_id = :") +
+              kStmtParamNameIndexId + NS_LITERAL_CSTRING(" AND value = :") +
+              kStmtParamNameValue + NS_LITERAL_CSTRING(";"),
           &stmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
@@ -18632,27 +18631,29 @@
     } else {
       rv = aConnection->GetCachedStatement(
           NS_LITERAL_CSTRING("DELETE FROM index_data "
-                             "WHERE index_id = :index_id "
-                             "AND value = :value "
-                             "AND object_data_key = :object_data_key;"),
+                             "WHERE index_id = :") +
+              kStmtParamNameIndexId + NS_LITERAL_CSTRING(" AND value = :") +
+              kStmtParamNameValue +
+              NS_LITERAL_CSTRING(" AND object_data_key = :") +
+              kStmtParamNameObjectDataKey + NS_LITERAL_CSTRING(";"),
           &stmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
     }
 
-    rv = stmt->BindInt64ByName(indexIdString, indexValue.mIndexId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = indexValue.mKey.BindToStatement(stmt, valueString);
+    rv = stmt->BindInt64ByName(kStmtParamNameIndexId, indexValue.mIndexId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = indexValue.mKey.BindToStatement(stmt, kStmtParamNameValue);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
 
     if (!indexValue.mUnique) {
-      rv = aObjectStoreKey.BindToStatement(stmt, objectDataKeyString);
+      rv = aObjectStoreKey.BindToStatement(stmt, kStmtParamNameObjectDataKey);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -18690,9 +18691,6 @@
 
   const bool singleRowOnly = aKeyRange.isSome() && aKeyRange.ref().isOnly();
 
-  NS_NAMED_LITERAL_CSTRING(objectStoreIdString, "object_store_id");
-  NS_NAMED_LITERAL_CSTRING(keyString, "key");
-
   nsresult rv;
   Key objectStoreKey;
   DatabaseConnection::CachedStatement selectStmt;
@@ -18701,8 +18699,9 @@
     rv = aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("SELECT index_data_values "
                            "FROM object_data "
-                           "WHERE object_store_id = :object_store_id "
-                           "AND key = :key;"),
+                           "WHERE object_store_id = :") +
+            kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(" AND key = :") +
+            kStmtParamNameKey + NS_LITERAL_CSTRING(";"),
         &selectStmt);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
@@ -18710,21 +18709,23 @@
 
     objectStoreKey = aKeyRange.ref().lower();
 
-    rv = objectStoreKey.BindToStatement(selectStmt, keyString);
+    rv = objectStoreKey.BindToStatement(selectStmt, kStmtParamNameKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
   } else {
     nsAutoCString keyRangeClause;
     if (aKeyRange.isSome()) {
-      GetBindingClauseForKeyRange(aKeyRange.ref(), keyString, keyRangeClause);
+      GetBindingClauseForKeyRange(aKeyRange.ref(), kStmtParamNameKey,
+                                  keyRangeClause);
     }
 
     rv = aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("SELECT index_data_values, key "
                            "FROM object_data "
                            "WHERE object_store_id = :") +
-            objectStoreIdString + keyRangeClause + NS_LITERAL_CSTRING(";"),
+            kStmtParamNameObjectStoreId + keyRangeClause +
+            NS_LITERAL_CSTRING(";"),
         &selectStmt);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
@@ -18738,7 +18739,7 @@
     }
   }
 
-  rv = selectStmt->BindInt64ByName(objectStoreIdString, aObjectStoreId);
+  rv = selectStmt->BindInt64ByName(kStmtParamNameObjectStoreId, aObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -18774,20 +18775,22 @@
     } else {
       rv = aConnection->GetCachedStatement(
           NS_LITERAL_CSTRING("DELETE FROM object_data "
-                             "WHERE object_store_id = :object_store_id "
-                             "AND key = :key;"),
+                             "WHERE object_store_id = :") +
+              kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(" AND key = :") +
+              kStmtParamNameKey + NS_LITERAL_CSTRING(";"),
           &deleteStmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
     }
 
-    rv = deleteStmt->BindInt64ByName(objectStoreIdString, aObjectStoreId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = objectStoreKey.BindToStatement(deleteStmt, keyString);
+    rv = deleteStmt->BindInt64ByName(kStmtParamNameObjectStoreId,
+                                     aObjectStoreId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = objectStoreKey.BindToStatement(deleteStmt, kStmtParamNameKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -18832,37 +18835,36 @@
   DatabaseConnection::CachedStatement updateStmt;
   rv = aConnection->GetCachedStatement(
       NS_LITERAL_CSTRING("UPDATE object_data "
-                         "SET index_data_values = :index_data_values "
-                         "WHERE object_store_id = :object_store_id "
-                         "AND key = :key;"),
+                         "SET index_data_values = :") +
+          kStmtParamNameIndexDataValues +
+          NS_LITERAL_CSTRING(" WHERE object_store_id = :") +
+          kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(" AND key = :") +
+          kStmtParamNameKey + NS_LITERAL_CSTRING(";"),
       &updateStmt);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
-  NS_NAMED_LITERAL_CSTRING(indexDataValuesString, "index_data_values");
-
   if (indexDataValues) {
-    rv = updateStmt->BindAdoptedBlobByName(indexDataValuesString,
+    rv = updateStmt->BindAdoptedBlobByName(kStmtParamNameIndexDataValues,
                                            indexDataValues.release(),
                                            indexDataValuesLength);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
   } else {
-    rv = updateStmt->BindNullByName(indexDataValuesString);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-  }
-
-  rv = updateStmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
-                                   aObjectStoreId);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = aObjectStoreKey.BindToStatement(updateStmt, NS_LITERAL_CSTRING("key"));
+    rv = updateStmt->BindNullByName(kStmtParamNameIndexDataValues);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+  }
+
+  rv = updateStmt->BindInt64ByName(kStmtParamNameObjectStoreId, aObjectStoreId);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = aObjectStoreKey.BindToStatement(updateStmt, kStmtParamNameKey);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -18889,15 +18891,14 @@
   nsresult rv = aConnection->GetCachedStatement(
       NS_LITERAL_CSTRING("SELECT id "
                          "FROM object_store_index "
-                         "WHERE object_store_id = :object_store_id "
-                         "LIMIT 1;"),
+                         "WHERE object_store_id = :") +
+          kStmtParamNameObjectStoreId + kOpenLimit + NS_LITERAL_CSTRING("1;"),
       &stmt);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
-                             aObjectStoreId);
+  rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, aObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -19513,18 +19514,12 @@
     if (aContentParent) {
       // Check to make sure that the child process has access to the database it
       // is accessing.
-      NS_NAMED_LITERAL_CSTRING(permissionStringBase,
-                               PERMISSION_STRING_CHROME_BASE);
       NS_ConvertUTF16toUTF8 databaseName(mCommonParams.metadata().name());
-      NS_NAMED_LITERAL_CSTRING(readSuffix,
-                               PERMISSION_STRING_CHROME_READ_SUFFIX);
-      NS_NAMED_LITERAL_CSTRING(writeSuffix,
-                               PERMISSION_STRING_CHROME_WRITE_SUFFIX);
 
       const nsAutoCString permissionStringWrite =
-          permissionStringBase + databaseName + writeSuffix;
+          kPermissionStringBase + databaseName + kPermissionWriteSuffix;
       const nsAutoCString permissionStringRead =
-          permissionStringBase + databaseName + readSuffix;
+          kPermissionStringBase + databaseName + kPermissionReadSuffix;
 
       bool canWrite = CheckAtLeastOneAppHasPermission(aContentParent,
                                                       permissionStringWrite);
@@ -19749,7 +19744,7 @@
   nsAutoString filename;
   GetDatabaseFilename(databaseName, filename);
 
-  rv = dbFile->Append(filename + NS_LITERAL_STRING(".sqlite"));
+  rv = dbFile->Append(filename + kSQLiteSuffix);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -20052,8 +20047,7 @@
     return rv;
   }
 
-  rv = markerFile->Append(NS_LITERAL_STRING(IDB_DELETION_MARKER_FILE_PREFIX) +
-                          filename);
+  rv = markerFile->Append(kIdbDeletionMarkerFilePrefix + filename);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -20081,7 +20075,7 @@
     return rv;
   }
 
-  rv = dbFile->Append(filename + NS_LITERAL_STRING(".sqlite"));
+  rv = dbFile->Append(filename + kSQLiteSuffix);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -20104,9 +20098,7 @@
     return rv;
   }
 
-  const NS_ConvertASCIItoUTF16 filesSuffix(kFileManagerDirectoryNameSuffix);
-
-  rv = fmDirectory->Append(filename + filesSuffix);
+  rv = fmDirectory->Append(filename + kFileManagerDirectoryNameSuffix);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -20527,6 +20519,8 @@
     indexTable.AssignLiteral("index_data");
   }
 
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   nsCString readQuery =
       NS_LITERAL_CSTRING("SELECT value, object_data_key FROM ") + indexTable +
       NS_LITERAL_CSTRING(" WHERE index_id = :index_id");
@@ -20536,8 +20530,7 @@
     return rv;
   }
 
-  rv = readStmt->BindInt64ByName(NS_LITERAL_CSTRING("index_id"),
-                                 aIndexMetadata.id());
+  rv = readStmt->BindInt64ByIndex(0, aIndexMetadata.id());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -20548,12 +20541,14 @@
   while (NS_SUCCEEDED((rv = readStmt->ExecuteStep(&hasResult))) && hasResult) {
     if (needCreateWriteQuery) {
       needCreateWriteQuery = false;
-      nsCString writeQuery = NS_LITERAL_CSTRING("UPDATE ") + indexTable +
-                             NS_LITERAL_CSTRING(
-                                 "SET value_locale = :value_locale "
-                                 "WHERE index_id = :index_id AND "
-                                 "value = :value AND "
-                                 "object_data_key = :object_data_key");
+      nsCString writeQuery =
+          NS_LITERAL_CSTRING("UPDATE ") + indexTable +
+          NS_LITERAL_CSTRING("SET value_locale = :") +
+          kStmtParamNameValueLocale +
+          NS_LITERAL_CSTRING(" WHERE index_id = :") + kStmtParamNameIndexId +
+          NS_LITERAL_CSTRING(" AND value = :") + kStmtParamNameValue +
+          NS_LITERAL_CSTRING(" AND object_data_key = :") +
+          kStmtParamNameObjectDataKey;
       rv = aConnection->CreateStatement(writeQuery, getter_AddRefs(writeStmt));
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
@@ -20561,8 +20556,7 @@
     }
 
     mozStorageStatementScoper scoper(writeStmt);
-    rv = writeStmt->BindInt64ByName(NS_LITERAL_CSTRING("index_id"),
-                                    aIndexMetadata.id());
+    rv = writeStmt->BindInt64ByName(kStmtParamNameIndexId, aIndexMetadata.id());
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -20573,7 +20567,7 @@
       return rv;
     }
 
-    rv = oldKey.BindToStatement(writeStmt, NS_LITERAL_CSTRING("value"));
+    rv = oldKey.BindToStatement(writeStmt, kStmtParamNameValue);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -20586,8 +20580,7 @@
                  : NS_ERROR_DOM_INDEXEDDB_DATA_ERR;
     }
 
-    rv = newSortKey.BindToStatement(writeStmt,
-                                    NS_LITERAL_CSTRING("value_locale"));
+    rv = newSortKey.BindToStatement(writeStmt, kStmtParamNameValueLocale);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -20597,8 +20590,7 @@
       return rv;
     }
 
-    rv = objectKey.BindToStatement(writeStmt,
-                                   NS_LITERAL_CSTRING("object_data_key"));
+    rv = objectKey.BindToStatement(writeStmt, kStmtParamNameObjectDataKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -20609,6 +20601,8 @@
     }
   }
 
+  // The parameter names are not used, parameters are bound by index only
+  // locally in the same function.
   nsCString metaQuery = NS_LITERAL_CSTRING(
       "UPDATE object_store_index SET "
       "locale = :locale WHERE id = :id");
@@ -20620,12 +20614,12 @@
 
   nsString locale;
   CopyASCIItoUTF16(aLocale, locale);
-  rv = metaStmt->BindStringByName(NS_LITERAL_CSTRING("locale"), locale);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = metaStmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), aIndexMetadata.id());
+  rv = metaStmt->BindStringByIndex(0, locale);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = metaStmt->BindInt64ByIndex(1, aIndexMetadata.id());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -21182,8 +21176,7 @@
     return rv;
   }
 
-  rv = updateStmt->BindInt64ByName(NS_LITERAL_CSTRING("version"),
-                                   int64_t(mRequestedVersion));
+  rv = updateStmt->BindInt64ByIndex(0, int64_t(mRequestedVersion));
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -21379,7 +21372,7 @@
     return rv;
   }
 
-  rv = dbFile->Append(filename + NS_LITERAL_STRING(".sqlite"));
+  rv = dbFile->Append(filename + kSQLiteSuffix);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -22015,9 +22008,6 @@
       mTransaction->mModifiedAutoIncrementObjectStoreMetadataArray;
 
   if (!metadataArray.IsEmpty()) {
-    NS_NAMED_LITERAL_CSTRING(osid, "osid");
-    NS_NAMED_LITERAL_CSTRING(ai, "ai");
-
     Database* database = mTransaction->GetDatabase();
     MOZ_ASSERT(database);
 
@@ -22036,23 +22026,24 @@
       if (stmt) {
         MOZ_ALWAYS_SUCCEEDS(stmt->Reset());
       } else {
+        // The parameter names are not used, parameters are bound by index only
+        // locally in the same function.
         rv = connection->GetCachedStatement(
             NS_LITERAL_CSTRING("UPDATE object_store "
-                               "SET auto_increment = :") +
-                ai + NS_LITERAL_CSTRING(" WHERE id = :") + osid +
-                NS_LITERAL_CSTRING(";"),
+                               "SET auto_increment = :auto_increment WHERE id "
+                               "= :object_store_id;"),
             &stmt);
         if (NS_WARN_IF(NS_FAILED(rv))) {
           return rv;
         }
       }
 
-      rv = stmt->BindInt64ByName(osid, metadata->mCommonMetadata.id());
-      if (NS_WARN_IF(NS_FAILED(rv))) {
-        return rv;
-      }
-
-      rv = stmt->BindInt64ByName(ai, metadata->mNextAutoIncrementId);
+      rv = stmt->BindInt64ByIndex(1, metadata->mCommonMetadata.id());
+      if (NS_WARN_IF(NS_FAILED(rv))) {
+        return rv;
+      }
+
+      rv = stmt->BindInt64ByIndex(0, metadata->mNextAutoIncrementId);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -22509,14 +22500,15 @@
     // another that already exists. This should be impossible because we should
     // have thrown an error long before now...
     DatabaseConnection::CachedStatement stmt;
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     MOZ_ALWAYS_SUCCEEDS(aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("SELECT name "
                            "FROM object_store "
                            "WHERE name = :name;"),
         &stmt));
 
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mMetadata.name()));
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindStringByIndex(0, mMetadata.name()));
 
     bool hasResult;
     MOZ_ALWAYS_SUCCEEDS(stmt->ExecuteStep(&hasResult));
@@ -22545,31 +22537,28 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mMetadata.id());
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt32ByName(NS_LITERAL_CSTRING("auto_increment"),
-                             mMetadata.autoIncrement() ? 1 : 0);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mMetadata.name());
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  NS_NAMED_LITERAL_CSTRING(keyPath, "key_path");
+  rv = stmt->BindInt64ByIndex(0, mMetadata.id());
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt32ByIndex(1, mMetadata.autoIncrement() ? 1 : 0);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindStringByIndex(2, mMetadata.name());
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
 
   if (mMetadata.keyPath().IsValid()) {
     nsAutoString keyPathSerialization;
     mMetadata.keyPath().SerializeToString(keyPathSerialization);
 
-    rv = stmt->BindStringByName(keyPath, keyPathSerialization);
-  } else {
-    rv = stmt->BindNullByName(keyPath);
+    rv = stmt->BindStringByIndex(3, keyPathSerialization);
+  } else {
+    rv = stmt->BindNullByIndex(3);
   }
 
   if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -22604,8 +22593,6 @@
 
   AUTO_PROFILER_LABEL("DeleteObjectStoreOp::DoDatabaseWork", DOM);
 
-  NS_NAMED_LITERAL_CSTRING(objectStoreIdString, "object_store_id");
-
 #ifdef DEBUG
   {
     // Make sure |mIsLastObjectStore| is telling the truth.
@@ -22730,13 +22717,14 @@
       DatabaseConnection::CachedStatement stmt;
       rv = aConnection->GetCachedStatement(
           NS_LITERAL_CSTRING("DELETE FROM object_store_index "
-                             "WHERE object_store_id = :object_store_id;"),
+                             "WHERE object_store_id = :") +
+              kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(";"),
           &stmt);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
 
-      rv = stmt->BindInt64ByName(objectStoreIdString,
+      rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId,
                                  mMetadata->mCommonMetadata.id());
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
@@ -22758,8 +22746,7 @@
         return rv;
       }
 
-      rv = stmt->BindInt64ByName(objectStoreIdString,
-                                 mMetadata->mCommonMetadata.id());
+      rv = stmt->BindInt64ByIndex(0, mMetadata->mCommonMetadata.id());
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -22779,8 +22766,7 @@
       return rv;
     }
 
-    rv = stmt->BindInt64ByName(objectStoreIdString,
-                               mMetadata->mCommonMetadata.id());
+    rv = stmt->BindInt64ByIndex(0, mMetadata->mCommonMetadata.id());
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -22824,17 +22810,17 @@
     // another that already exists. This should be impossible because we should
     // have thrown an error long before now...
     DatabaseConnection::CachedStatement stmt;
-    MOZ_ALWAYS_SUCCEEDS(
-        aConnection->GetCachedStatement(NS_LITERAL_CSTRING("SELECT name "
-                                                           "FROM object_store "
-                                                           "WHERE name = :name "
-                                                           "AND id != :id;"),
-                                        &stmt));
-
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mNewName));
-
-    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mId));
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
+    MOZ_ALWAYS_SUCCEEDS(aConnection->GetCachedStatement(
+        NS_LITERAL_CSTRING("SELECT name "
+                           "FROM object_store "
+                           "WHERE name = :name AND id != :id;"),
+        &stmt));
+
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindStringByIndex(0, mNewName));
+
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByIndex(1, mId));
 
     bool hasResult;
     MOZ_ALWAYS_SUCCEEDS(stmt->ExecuteStep(&hasResult));
@@ -22862,13 +22848,13 @@
     return rv;
   }
 
-  rv = stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mNewName);
-
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mId);
+  rv = stmt->BindStringByIndex(0, mNewName);
+
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt64ByIndex(1, mId);
 
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -22955,8 +22941,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
-                             mObjectStoreId);
+  rv = stmt->BindInt64ByIndex(0, mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -22994,16 +22979,16 @@
     // store as another that already exists. This should be impossible because
     // we should have thrown an error long before now...
     DatabaseConnection::CachedStatement stmt;
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     MOZ_ALWAYS_SUCCEEDS(aConnection->GetCachedStatement(
-        NS_LITERAL_CSTRING("SELECT name "
-                           "FROM object_store_index "
-                           "WHERE object_store_id = :osid "
-                           "AND name = :name;"),
+        NS_LITERAL_CSTRING(
+            "SELECT name "
+            "FROM object_store_index "
+            "WHERE object_store_id = :object_store_id AND name = :name;"),
         &stmt));
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"), mObjectStoreId));
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mMetadata.name()));
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByIndex(0, mObjectStoreId));
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindStringByIndex(1, mMetadata.name()));
 
     bool hasResult;
     MOZ_ALWAYS_SUCCEEDS(stmt->ExecuteStep(&hasResult));
@@ -23029,61 +23014,55 @@
           "INSERT INTO object_store_index (id, name, key_path, unique_index, "
           "multientry, object_store_id, locale, "
           "is_auto_locale) "
-          "VALUES (:id, :name, :key_path, :unique, :multientry, :osid, "
-          ":locale, "
-          ":is_auto_locale)"),
+          "VALUES (:id, :name, :key_path, :unique, :multientry, "
+          ":object_store_id, :locale, :is_auto_locale)"),
       &stmt);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mMetadata.id());
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mMetadata.name());
+  rv = stmt->BindInt64ByIndex(0, mMetadata.id());
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindStringByIndex(1, mMetadata.name());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   nsAutoString keyPathSerialization;
   mMetadata.keyPath().SerializeToString(keyPathSerialization);
-  rv = stmt->BindStringByName(NS_LITERAL_CSTRING("key_path"),
-                              keyPathSerialization);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt32ByName(NS_LITERAL_CSTRING("unique"),
-                             mMetadata.unique() ? 1 : 0);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt32ByName(NS_LITERAL_CSTRING("multientry"),
-                             mMetadata.multiEntry() ? 1 : 0);
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"), mObjectStoreId);
+  rv = stmt->BindStringByIndex(2, keyPathSerialization);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt32ByIndex(3, mMetadata.unique() ? 1 : 0);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt32ByIndex(4, mMetadata.multiEntry() ? 1 : 0);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt64ByIndex(5, mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   if (mMetadata.locale().IsEmpty()) {
-    rv = stmt->BindNullByName(NS_LITERAL_CSTRING("locale"));
-  } else {
-    rv = stmt->BindUTF8StringByName(NS_LITERAL_CSTRING("locale"),
-                                    mMetadata.locale());
-  }
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt32ByName(NS_LITERAL_CSTRING("is_auto_locale"),
-                             mMetadata.autoLocale());
+    rv = stmt->BindNullByIndex(6);
+  } else {
+    rv = stmt->BindUTF8StringByIndex(6, mMetadata.locale());
+  }
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt32ByIndex(7, mMetadata.autoLocale());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -23345,20 +23324,20 @@
     nsresult rv = aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("UPDATE object_data "
                            "SET index_data_values = NULL "
-                           "WHERE object_store_id = :object_store_id "
-                           "AND key = :key;"),
+                           "WHERE object_store_id = :") +
+            kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(" AND key = :") +
+            kStmtParamNameKey + NS_LITERAL_CSTRING(";"),
         &stmt);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
 
-    rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
-                               mObjectStoreId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = aObjectStoreKey.BindToStatement(stmt, NS_LITERAL_CSTRING("key"));
+    rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, mObjectStoreId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = aObjectStoreKey.BindToStatement(stmt, kStmtParamNameKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -23435,14 +23414,15 @@
   {
     // Make sure |mIsLastIndex| is telling the truth.
     DatabaseConnection::CachedStatement stmt;
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     MOZ_ALWAYS_SUCCEEDS(aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("SELECT id "
                            "FROM object_store_index "
                            "WHERE object_store_id = :object_store_id;"),
         &stmt));
 
-    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByName(
-        NS_LITERAL_CSTRING("object_store_id"), mObjectStoreId));
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByIndex(0, mObjectStoreId));
 
     bool foundThisIndex = false;
     bool foundOtherIndex = false;
@@ -23494,8 +23474,9 @@
           NS_LITERAL_CSTRING("/* do not warn (bug someone else) */ "
                              "SELECT value, object_data_key "
                              "FROM unique_index_data "
-                             "WHERE index_id = :index_id "
-                             "ORDER BY object_data_key ASC;"),
+                             "WHERE index_id = :") +
+              kStmtParamNameIndexId +
+              NS_LITERAL_CSTRING(" ORDER BY object_data_key ASC;"),
           &selectStmt);
     } else {
       rv = aConnection->GetCachedStatement(
@@ -23507,9 +23488,12 @@
               "FROM unique_index_data "
               "JOIN object_data "
               "ON unique_index_data.object_data_key = object_data.key "
-              "WHERE unique_index_data.index_id = :index_id "
-              "AND object_data.object_store_id = :object_store_id "
-              "ORDER BY unique_index_data.object_data_key ASC;"),
+              "WHERE unique_index_data.index_id = :") +
+              kStmtParamNameIndexId +
+              NS_LITERAL_CSTRING(" AND object_data.object_store_id = :") +
+              kStmtParamNameObjectStoreId +
+              NS_LITERAL_CSTRING(
+                  " ORDER BY unique_index_data.object_data_key ASC;"),
           &selectStmt);
     }
   } else {
@@ -23518,23 +23502,26 @@
           NS_LITERAL_CSTRING("/* do not warn (bug me not) */ "
                              "SELECT value, object_data_key "
                              "FROM index_data "
-                             "WHERE index_id = :index_id "
-                             "AND object_store_id = :object_store_id "
-                             "ORDER BY object_data_key ASC;"),
+                             "WHERE index_id = :") +
+              kStmtParamNameIndexId +
+              NS_LITERAL_CSTRING(" AND object_store_id = :") +
+              kStmtParamNameObjectStoreId +
+              NS_LITERAL_CSTRING(" ORDER BY object_data_key ASC;"),
           &selectStmt);
     } else {
       rv = aConnection->GetCachedStatement(
-          NS_LITERAL_CSTRING(
-              "/* do not warn (bug off) */ "
-              "SELECT index_data.value, "
-              "index_data.object_data_key, "
-              "object_data.index_data_values "
-              "FROM index_data "
-              "JOIN object_data "
-              "ON index_data.object_data_key = object_data.key "
-              "WHERE index_data.index_id = :index_id "
-              "AND object_data.object_store_id = :object_store_id "
-              "ORDER BY index_data.object_data_key ASC;"),
+          NS_LITERAL_CSTRING("/* do not warn (bug off) */ "
+                             "SELECT index_data.value, "
+                             "index_data.object_data_key, "
+                             "object_data.index_data_values "
+                             "FROM index_data "
+                             "JOIN object_data "
+                             "ON index_data.object_data_key = object_data.key "
+                             "WHERE index_data.index_id = :") +
+              kStmtParamNameIndexId +
+              NS_LITERAL_CSTRING(" AND object_data.object_store_id = :") +
+              kStmtParamNameObjectStoreId +
+              NS_LITERAL_CSTRING(" ORDER BY index_data.object_data_key ASC;"),
           &selectStmt);
     }
   }
@@ -23542,24 +23529,19 @@
     return rv;
   }
 
-  NS_NAMED_LITERAL_CSTRING(indexIdString, "index_id");
-
-  rv = selectStmt->BindInt64ByName(indexIdString, mIndexId);
+  rv = selectStmt->BindInt64ByName(kStmtParamNameIndexId, mIndexId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   if (!mUnique || !mIsLastIndex) {
-    rv = selectStmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
+    rv = selectStmt->BindInt64ByName(kStmtParamNameObjectStoreId,
                                      mObjectStoreId);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
   }
 
-  NS_NAMED_LITERAL_CSTRING(valueString, "value");
-  NS_NAMED_LITERAL_CSTRING(objectDataKeyString, "object_data_key");
-
   DatabaseConnection::CachedStatement deleteIndexRowStmt;
   DatabaseConnection::CachedStatement nullIndexDataValuesStmt;
 
@@ -23635,15 +23617,18 @@
       if (mUnique) {
         rv = aConnection->GetCachedStatement(
             NS_LITERAL_CSTRING("DELETE FROM unique_index_data "
-                               "WHERE index_id = :index_id "
-                               "AND value = :value;"),
+                               "WHERE index_id = :") +
+                kStmtParamNameIndexId + NS_LITERAL_CSTRING(" AND value = :") +
+                kStmtParamNameValue + NS_LITERAL_CSTRING(";"),
             &deleteIndexRowStmt);
       } else {
         rv = aConnection->GetCachedStatement(
             NS_LITERAL_CSTRING("DELETE FROM index_data "
-                               "WHERE index_id = :index_id "
-                               "AND value = :value "
-                               "AND object_data_key = :object_data_key;"),
+                               "WHERE index_id = :") +
+                kStmtParamNameIndexId + NS_LITERAL_CSTRING(" AND value = :") +
+                kStmtParamNameValue +
+                NS_LITERAL_CSTRING(" AND object_data_key = :") +
+                kStmtParamNameObjectDataKey + NS_LITERAL_CSTRING(";"),
             &deleteIndexRowStmt);
       }
       if (NS_WARN_IF(NS_FAILED(rv))) {
@@ -23651,19 +23636,19 @@
       }
     }
 
-    rv = deleteIndexRowStmt->BindInt64ByName(indexIdString, mIndexId);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-
-    rv = indexKey.BindToStatement(deleteIndexRowStmt, valueString);
+    rv = deleteIndexRowStmt->BindInt64ByName(kStmtParamNameIndexId, mIndexId);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+
+    rv = indexKey.BindToStatement(deleteIndexRowStmt, kStmtParamNameValue);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
 
     if (!mUnique) {
       rv = lastObjectStoreKey.BindToStatement(deleteIndexRowStmt,
-                                              objectDataKeyString);
+                                              kStmtParamNameObjectDataKey);
       if (NS_WARN_IF(NS_FAILED(rv))) {
         return rv;
       }
@@ -23698,7 +23683,7 @@
     return rv;
   }
 
-  rv = deleteStmt->BindInt64ByName(indexIdString, mIndexId);
+  rv = deleteStmt->BindInt64ByIndex(0, mIndexId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -23737,6 +23722,8 @@
     // that already exists. This should be impossible because we should have
     // thrown an error long before now...
     DatabaseConnection::CachedStatement stmt;
+    // The parameter names are not used, parameters are bound by index only
+    // locally in the same function.
     MOZ_ALWAYS_SUCCEEDS(aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("SELECT name "
                            "FROM object_store_index "
@@ -23745,14 +23732,11 @@
                            "AND id != :id;"),
         &stmt));
 
-    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByName(
-        NS_LITERAL_CSTRING("object_store_id"), mObjectStoreId));
-
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mNewName));
-
-    MOZ_ALWAYS_SUCCEEDS(
-        stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mIndexId));
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByIndex(0, mObjectStoreId));
+
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindStringByIndex(1, mNewName));
+
+    MOZ_ALWAYS_SUCCEEDS(stmt->BindInt64ByIndex(2, mIndexId));
 
     bool hasResult;
     MOZ_ALWAYS_SUCCEEDS(stmt->ExecuteStep(&hasResult));
@@ -23783,13 +23767,13 @@
     return rv;
   }
 
-  rv = stmt->BindStringByName(NS_LITERAL_CSTRING("name"), mNewName);
-
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), mIndexId);
+  rv = stmt->BindStringByIndex(0, mNewName);
+
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt64ByIndex(1, mIndexId);
 
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -24031,20 +24015,21 @@
   nsresult rv = aConnection->GetCachedStatement(
       NS_LITERAL_CSTRING("SELECT index_data_values "
                          "FROM object_data "
-                         "WHERE object_store_id = :object_store_id "
-                         "AND key = :key;"),
+                         "WHERE object_store_id = :") +
+          kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(" AND key = :") +
+          kStmtParamNameKey + NS_LITERAL_CSTRING(";"),
       &indexValuesStmt);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
-  rv = indexValuesStmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
+  rv = indexValuesStmt->BindInt64ByName(kStmtParamNameObjectStoreId,
                                         mParams.objectStoreId());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
-  rv = mResponse.BindToStatement(indexValuesStmt, NS_LITERAL_CSTRING("key"));
+  rv = mResponse.BindToStatement(indexValuesStmt, kStmtParamNameKey);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -24236,24 +24221,24 @@
   // if we allow overwrite or not. By not allowing overwrite we raise
   // detectable errors rather than corrupting data.
   DatabaseConnection::CachedStatement stmt;
-  if (!mOverwrite || keyUnset) {
-    rv = aConnection->GetCachedStatement(
-        NS_LITERAL_CSTRING("INSERT INTO object_data "
-                           "(object_store_id, key, file_ids, data) "
-                           "VALUES (:osid, :key, :file_ids, :data);"),
-        &stmt);
-  } else {
-    rv = aConnection->GetCachedStatement(
-        NS_LITERAL_CSTRING("INSERT OR REPLACE INTO object_data "
-                           "(object_store_id, key, file_ids, data) "
-                           "VALUES (:osid, :key, :file_ids, :data);"),
-        &stmt);
-  }
-  if (NS_WARN_IF(NS_FAILED(rv))) {
-    return rv;
-  }
-
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"), osid);
+  const auto& optReplaceDirective = (!mOverwrite || keyUnset)
+                                        ? NS_LITERAL_CSTRING("")
+                                        : NS_LITERAL_CSTRING("OR REPLACE ");
+  rv = aConnection->GetCachedStatement(
+      NS_LITERAL_CSTRING("INSERT ") + optReplaceDirective +
+          NS_LITERAL_CSTRING("INTO object_data "
+                             "(object_store_id, key, file_ids, data) "
+                             "VALUES (:") +
+          kStmtParamNameObjectStoreId + NS_LITERAL_CSTRING(", :") +
+          kStmtParamNameKey + NS_LITERAL_CSTRING(", :") +
+          kStmtParamNameFileIds + NS_LITERAL_CSTRING(", :") +
+          kStmtParamNameData + NS_LITERAL_CSTRING(");"),
+      &stmt);
+  if (NS_WARN_IF(NS_FAILED(rv))) {
+    return rv;
+  }
+
+  rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, osid);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -24311,7 +24296,7 @@
     }
   }
 
-  key.BindToStatement(stmt, NS_LITERAL_CSTRING("key"));
+  key.BindToStatement(stmt, kStmtParamNameKey);
 
   if (mDataOverThreshold) {
     // The data we store in the SQLite database is a (signed) 64-bit integer.
@@ -24326,7 +24311,7 @@
 
     int64_t data = (uint64_t(flags) << 32) | index;
 
-    rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("data"), data);
+    rv = stmt->BindInt64ByName(kStmtParamNameData, data);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -24357,7 +24342,7 @@
     uint8_t* dataBuffer = reinterpret_cast<uint8_t*>(compressed.release());
     size_t dataBufferLength = compressedLength;
 
-    rv = stmt->BindAdoptedBlobByName(NS_LITERAL_CSTRING("data"), dataBuffer,
+    rv = stmt->BindAdoptedBlobByName(kStmtParamNameData, dataBuffer,
                                      dataBufferLength);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
@@ -24455,12 +24440,12 @@
       storedFileInfo.Serialize(fileIds);
     }
 
-    rv = stmt->BindStringByName(NS_LITERAL_CSTRING("file_ids"), fileIds);
-    if (NS_WARN_IF(NS_FAILED(rv))) {
-      return rv;
-    }
-  } else {
-    rv = stmt->BindNullByName(NS_LITERAL_CSTRING("file_ids"));
+    rv = stmt->BindStringByName(kStmtParamNameFileIds, fileIds);
+    if (NS_WARN_IF(NS_FAILED(rv))) {
+      return rv;
+    }
+  } else {
+    rv = stmt->BindNullByName(kStmtParamNameFileIds);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -24662,22 +24647,22 @@
 
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(),
-                                NS_LITERAL_CSTRING("key"), keyRangeClause);
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameKey,
+                                keyRangeClause);
   }
 
   nsCString limitClause;
   if (mLimit) {
-    limitClause.AssignLiteral(" LIMIT ");
+    limitClause = kOpenLimit;
     limitClause.AppendInt(mLimit);
   }
 
   nsCString query = NS_LITERAL_CSTRING(
                         "SELECT file_ids, data "
                         "FROM object_data "
-                        "WHERE object_store_id = :osid") +
-                    keyRangeClause + NS_LITERAL_CSTRING(" ORDER BY key ASC") +
-                    limitClause;
+                        "WHERE object_store_id = :") +
+                    kStmtParamNameObjectStoreId + keyRangeClause +
+                    NS_LITERAL_CSTRING(" ORDER BY key ASC") + limitClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -24685,7 +24670,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"), mObjectStoreId);
+  rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -24856,8 +24841,8 @@
 
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(),
-                                NS_LITERAL_CSTRING("key"), keyRangeClause);
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameKey,
+                                keyRangeClause);
   }
 
   nsAutoCString limitClause;
@@ -24869,9 +24854,9 @@
   nsCString query = NS_LITERAL_CSTRING(
                         "SELECT key "
                         "FROM object_data "
-                        "WHERE object_store_id = :osid") +
-                    keyRangeClause + NS_LITERAL_CSTRING(" ORDER BY key ASC") +
-                    limitClause;
+                        "WHERE object_store_id = :") +
+                    kStmtParamNameObjectStoreId + keyRangeClause +
+                    NS_LITERAL_CSTRING(" ORDER BY key ASC") + limitClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -24879,7 +24864,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"), mObjectStoreId);
+  rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId, mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -24990,23 +24975,23 @@
       return rv;
     }
   } else {
-    NS_NAMED_LITERAL_CSTRING(objectStoreIdString, "object_store_id");
-
     nsAutoCString keyRangeClause;
-    GetBindingClauseForKeyRange(mParams.keyRange(), NS_LITERAL_CSTRING("key"),
+    GetBindingClauseForKeyRange(mParams.keyRange(), kColumnNameKey,
                                 keyRangeClause);
 
     DatabaseConnection::CachedStatement stmt;
     rv = aConnection->GetCachedStatement(
         NS_LITERAL_CSTRING("DELETE FROM object_data "
                            "WHERE object_store_id = :") +
-            objectStoreIdString + keyRangeClause + NS_LITERAL_CSTRING(";"),
+            kStmtParamNameObjectStoreId + keyRangeClause +
+            NS_LITERAL_CSTRING(";"),
         &stmt);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
 
-    rv = stmt->BindInt64ByName(objectStoreIdString, mParams.objectStoreId());
+    rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId,
+                               mParams.objectStoreId());
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -25087,8 +25072,7 @@
       return rv;
     }
 
-    rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("object_store_id"),
-                               mParams.objectStoreId());
+    rv = stmt->BindInt64ByIndex(0, mParams.objectStoreId());
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -25119,14 +25103,14 @@
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
     GetBindingClauseForKeyRange(mParams.optionalKeyRange().ref(),
-                                NS_LITERAL_CSTRING("key"), keyRangeClause);
+                                kColumnNameKey, keyRangeClause);
   }
 
   nsCString query = NS_LITERAL_CSTRING(
                         "SELECT count(*) "
                         "FROM object_data "
-                        "WHERE object_store_id = :osid") +
-                    keyRangeClause;
+                        "WHERE object_store_id = :") +
+                    kStmtParamNameObjectStoreId + keyRangeClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -25134,7 +25118,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("osid"),
+  rv = stmt->BindInt64ByName(kStmtParamNameObjectStoreId,
                              mParams.objectStoreId());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -25272,13 +25256,13 @@
 
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(),
-                                NS_LITERAL_CSTRING("value"), keyRangeClause);
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameValue,
+                                keyRangeClause);
   }
 
   nsCString limitClause;
   if (mLimit) {
-    limitClause.AssignLiteral(" LIMIT ");
+    limitClause = kOpenLimit;
     limitClause.AppendInt(mLimit);
   }
 
@@ -25293,8 +25277,8 @@
                         "index_table.object_store_id "
                         "AND object_data.key = "
                         "index_table.object_data_key "
-                        "WHERE index_id = :index_id") +
-                    keyRangeClause + limitClause;
+                        "WHERE index_id = :") +
+                    kStmtParamNameIndexId + keyRangeClause + limitClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -25302,7 +25286,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("index_id"),
+  rv = stmt->BindInt64ByName(kStmtParamNameIndexId,
                              mMetadata->mCommonMetadata.id());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -25452,22 +25436,21 @@
 
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(),
-                                NS_LITERAL_CSTRING("value"), keyRangeClause);
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameValue,
+                                keyRangeClause);
   }
 
   nsCString limitClause;
   if (mLimit) {
-    limitClause.AssignLiteral(" LIMIT ");
+    limitClause = kOpenLimit;
     limitClause.AppendInt(mLimit);
   }
 
   nsCString query = NS_LITERAL_CSTRING(
                         "SELECT object_data_key "
                         "FROM ") +
-                    indexTable +
-                    NS_LITERAL_CSTRING("WHERE index_id = :index_id") +
-                    keyRangeClause + limitClause;
+                    indexTable + NS_LITERAL_CSTRING("WHERE index_id = :") +
+                    kStmtParamNameIndexId + keyRangeClause + limitClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -25475,7 +25458,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("index_id"),
+  rv = stmt->BindInt64ByName(kStmtParamNameIndexId,
                              mMetadata->mCommonMetadata.id());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -25555,15 +25538,14 @@
   nsAutoCString keyRangeClause;
   if (hasKeyRange) {
     GetBindingClauseForKeyRange(mParams.optionalKeyRange().ref(),
-                                NS_LITERAL_CSTRING("value"), keyRangeClause);
+                                kColumnNameValue, keyRangeClause);
   }
 
   nsCString query = NS_LITERAL_CSTRING(
                         "SELECT count(*) "
                         "FROM ") +
-                    indexTable +
-                    NS_LITERAL_CSTRING("WHERE index_id = :index_id") +
-                    keyRangeClause;
+                    indexTable + NS_LITERAL_CSTRING("WHERE index_id = :") +
+                    kStmtParamNameIndexId + keyRangeClause;
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(query, &stmt);
@@ -25571,7 +25553,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("index_id"),
+  rv = stmt->BindInt64ByName(kStmtParamNameIndexId,
                              mMetadata->mCommonMetadata.id());
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
@@ -25803,13 +25785,12 @@
     const nsACString& aQueryStart) {
   const bool isIncreasingOrder = IsIncreasingOrder(mCursor->mDirection);
 
-  NS_NAMED_LITERAL_CSTRING(currentKey, "current_key");
   nsAutoCString keyRangeClause;
   nsAutoCString continueToKeyRangeClause;
-  AppendConditionClause(aKeyString, currentKey, !isIncreasingOrder, false,
-                        keyRangeClause);
-  AppendConditionClause(aKeyString, currentKey, !isIncreasingOrder, true,
-                        continueToKeyRangeClause);
+  AppendConditionClause(aKeyString, kStmtParamNameCurrentKey,
+                        !isIncreasingOrder, false, keyRangeClause);
+  AppendConditionClause(aKeyString, kStmtParamNameCurrentKey,
+                        !isIncreasingOrder, true, continueToKeyRangeClause);
 
   {
     Key bound;
@@ -25817,31 +25798,24 @@
     GetRangeKeyInfo(!isIncreasingOrder, &bound, &open);
 
     if (mOptionalKeyRange.isSome() && !bound.IsUnset()) {
-      NS_NAMED_LITERAL_CSTRING(rangeKey, "range_key");
-
-      AppendConditionClause(aKeyString, rangeKey, isIncreasingOrder, !open,
-                            keyRangeClause);
-      AppendConditionClause(aKeyString, rangeKey, isIncreasingOrder, !open,
-                            continueToKeyRangeClause);
+      AppendConditionClause(aKeyString, kStmtParamNameRangeKey,
+                            isIncreasingOrder, !open, keyRangeClause);
+      AppendConditionClause(aKeyString, kStmtParamNameRangeKey,
+                            isIncreasingOrder, !open, continueToKeyRangeClause);
       mCursor->mRangeKey = std::move(bound);
     }
   }
 
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
-
   mCursor->mContinueQuery =
-      aQueryStart + keyRangeClause + aDirectionClause + openLimit;
+      aQueryStart + keyRangeClause + aDirectionClause + kOpenLimit;
 
   mCursor->mContinueToQuery =
-      aQueryStart + continueToKeyRangeClause + aDirectionClause + openLimit;
+      aQueryStart + continueToKeyRangeClause + aDirectionClause + kOpenLimit;
 }
 
 void Cursor::OpenOp::PrepareIndexKeyConditionClause(
     const nsACString& aSortColumn, const nsACString& aDirectionClause,
     const nsLiteralCString& aObjectDataKeyPrefix, nsAutoCString aQueryStart) {
-  NS_NAMED_LITERAL_CSTRING(rangeKey, "range_key");
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
-
   const bool isIncreasingOrder = IsIncreasingOrder(mCursor->mDirection);
 
   {
@@ -25849,8 +25823,8 @@
     bool open;
     GetRangeKeyInfo(!isIncreasingOrder, &bound, &open);
     if (mOptionalKeyRange.isSome() && !bound.IsUnset()) {
-      AppendConditionClause(aSortColumn, rangeKey, isIncreasingOrder, !open,
-                            aQueryStart);
+      AppendConditionClause(aSortColumn, kStmtParamNameRangeKey,
+                            isIncreasingOrder, !open, aQueryStart);
       mCursor->mRangeKey = std::move(bound);
     }
   }
@@ -25860,45 +25834,44 @@
 
   mCursor->mContinueToQuery =
       aQueryStart + NS_LITERAL_CSTRING(" AND sort_column ") + comparisonChar +
-      NS_LITERAL_CSTRING("= :current_key") + aDirectionClause + openLimit;
+      NS_LITERAL_CSTRING("= :") + kStmtParamNameCurrentKey + aDirectionClause +
+      kOpenLimit;
 
   switch (mCursor->mDirection) {
     case IDBCursor::NEXT:
     case IDBCursor::PREV:
       mCursor->mContinueQuery =
           aQueryStart + NS_LITERAL_CSTRING(" AND sort_column ") +
-          comparisonChar +
-          NS_LITERAL_CSTRING(
-              "= :current_key "
-              "AND ( sort_column ") +
-          comparisonChar + NS_LITERAL_CSTRING(" :current_key OR ") +
-          aObjectDataKeyPrefix + NS_LITERAL_CSTRING("object_data_key ") +
-          comparisonChar + NS_LITERAL_CSTRING(" :object_key ) ") +
-          aDirectionClause + openLimit;
+          comparisonChar + NS_LITERAL_CSTRING("= :") +
+          kStmtParamNameCurrentKey + NS_LITERAL_CSTRING(" AND ( sort_column ") +
+          comparisonChar + NS_LITERAL_CSTRING(" :") + kStmtParamNameCurrentKey +
+          NS_LITERAL_CSTRING(" OR ") + aObjectDataKeyPrefix +
+          NS_LITERAL_CSTRING("object_data_key ") + comparisonChar +
+          NS_LITERAL_CSTRING(" :") + kStmtParamNameObjectKey +
+          NS_LITERAL_CSTRING(" ) ") + aDirectionClause + kOpenLimit;
 
       mCursor->mContinuePrimaryKeyQuery =
           aQueryStart +
           NS_LITERAL_CSTRING(
               " AND ("
-              "(sort_column == :current_key AND ") +
+              "(sort_column == :") +
+          kStmtParamNameCurrentKey + NS_LITERAL_CSTRING(" AND ") +
           aObjectDataKeyPrefix + NS_LITERAL_CSTRING("object_data_key ") +
-          comparisonChar +
+          comparisonChar + NS_LITERAL_CSTRING("= :") + kStmtParamNameObjectKey +
           NS_LITERAL_CSTRING(
-              "= :object_key) OR "
+              ") OR "
               "sort_column ") +
-          comparisonChar +
-          NS_LITERAL_CSTRING(
-              " :current_key"
-              ")") +
-          aDirectionClause + openLimit;
+          comparisonChar + NS_LITERAL_CSTRING(" :") + kStmtParamNameCurrentKey +
+          kStmtParamNameCurrentKey + NS_LITERAL_CSTRING(")") +
+          aDirectionClause + kOpenLimit;
       break;
 
     case IDBCursor::NEXT_UNIQUE:
     case IDBCursor::PREV_UNIQUE:
       mCursor->mContinueQuery =
           aQueryStart + NS_LITERAL_CSTRING(" AND sort_column ") +
-          comparisonChar + NS_LITERAL_CSTRING(" :current_key") +
-          aDirectionClause + openLimit;
+          comparisonChar + NS_LITERAL_CSTRING(" :") + kStmtParamNameCurrentKey +
+          aDirectionClause + kOpenLimit;
       break;
 
     default:
@@ -25919,24 +25892,21 @@
 
   const bool usingKeyRange = mOptionalKeyRange.isSome();
 
-  NS_NAMED_LITERAL_CSTRING(keyString, "key");
-  NS_NAMED_LITERAL_CSTRING(id, "id");
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
-
-  nsCString queryStart = NS_LITERAL_CSTRING("SELECT ") + keyString +
+  nsCString queryStart = NS_LITERAL_CSTRING("SELECT ") + kColumnNameKey +
                          NS_LITERAL_CSTRING(
                              ", file_ids, data "
                              "FROM object_data "
                              "WHERE object_store_id = :") +
-                         id;
+                         kStmtParamNameId;
 
   nsAutoCString keyRangeClause;
   if (usingKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), keyString,
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameKey,
                                 keyRangeClause);
   }
 
-  nsAutoCString directionClause = NS_LITERAL_CSTRING(" ORDER BY ") + keyString;
+  nsAutoCString directionClause =
+      NS_LITERAL_CSTRING(" ORDER BY ") + kColumnNameKey;
   switch (mCursor->mDirection) {
     case IDBCursor::NEXT:
     case IDBCursor::NEXT_UNIQUE:
@@ -25954,8 +25924,8 @@
 
   // Note: Changing the number or order of SELECT columns in the query will
   // require changes to CursorOpBase::PopulateResponseFromStatement.
-  nsCString firstQuery = queryStart + keyRangeClause + directionClause +
-                         openLimit + NS_LITERAL_CSTRING("1");
+  const nsCString firstQuery = queryStart + keyRangeClause + directionClause +
+                               kOpenLimit + NS_LITERAL_CSTRING("1");
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(firstQuery, &stmt);
@@ -25963,7 +25933,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(id, mCursor->mObjectStoreId);
+  rv = stmt->BindInt64ByName(kStmtParamNameId, mCursor->mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -25992,7 +25962,7 @@
   }
 
   // Now we need to make the query to get the next match.
-  PrepareKeyConditionClauses(keyString, directionClause, queryStart);
+  PrepareKeyConditionClauses(kStmtParamNameKey, directionClause, queryStart);
 
   return NS_OK;
 }
@@ -26010,31 +25980,27 @@
 
   const bool usingKeyRange = mOptionalKeyRange.isSome();
 
-  NS_NAMED_LITERAL_CSTRING(keyString, "key");
-  NS_NAMED_LITERAL_CSTRING(id, "id");
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
-
-  nsCString queryStart = NS_LITERAL_CSTRING("SELECT ") + keyString +
+  nsCString queryStart = NS_LITERAL_CSTRING("SELECT ") + kColumnNameKey +
                          NS_LITERAL_CSTRING(
                              " FROM object_data "
                              "WHERE object_store_id = :") +
-                         id;
+                         kStmtParamNameId;
 
   nsAutoCString keyRangeClause;
   if (usingKeyRange) {
-    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), keyString,
+    GetBindingClauseForKeyRange(mOptionalKeyRange.ref(), kColumnNameKey,
                                 keyRangeClause);
   }
 
   const nsAutoCString directionClause =
-      NS_LITERAL_CSTRING(" ORDER BY ") + keyString +
+      NS_LITERAL_CSTRING(" ORDER BY ") + kColumnNameKey +
       (IsIncreasingOrder(mCursor->mDirection) ? NS_LITERAL_CSTRING(" ASC")
                                               : NS_LITERAL_CSTRING(" DESC"));
 
   // Note: Changing the number or order of SELECT columns in the query will
   // require changes to CursorOpBase::PopulateResponseFromStatement.
   nsCString firstQuery = queryStart + keyRangeClause + directionClause +
-                         openLimit + NS_LITERAL_CSTRING("1");
+                         kOpenLimit + NS_LITERAL_CSTRING("1");
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(firstQuery, &stmt);
@@ -26042,7 +26008,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(id, mCursor->mObjectStoreId);
+  rv = stmt->BindInt64ByName(kStmtParamNameId, mCursor->mObjectStoreId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -26071,7 +26037,7 @@
   }
 
   // Now we need to make the query to get the next match.
-  PrepareKeyConditionClauses(keyString, directionClause, queryStart);
+  PrepareKeyConditionClauses(kStmtParamNameKey, directionClause, queryStart);
 
   return NS_OK;
 }
@@ -26093,8 +26059,6 @@
                              : NS_LITERAL_CSTRING("index_data");
 
   NS_NAMED_LITERAL_CSTRING(sortColumn, "sort_column");
-  NS_NAMED_LITERAL_CSTRING(id, "id");
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
 
   nsAutoCString sortColumnAlias;
   if (mCursor->IsLocaleAware()) {
@@ -26122,7 +26086,7 @@
                                  "AND index_table.object_data_key = "
                                  "object_data.key "
                                  "WHERE index_table.index_id = :") +
-                             id;
+                             kStmtParamNameId;
 
   nsAutoCString keyRangeClause;
   if (usingKeyRange) {
@@ -26153,7 +26117,7 @@
   // Note: Changing the number or order of SELECT columns in the query will
   // require changes to CursorOpBase::PopulateResponseFromStatement.
   nsCString firstQuery = queryStart + keyRangeClause + directionClause +
-                         openLimit + NS_LITERAL_CSTRING("1");
+                         kOpenLimit + NS_LITERAL_CSTRING("1");
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(firstQuery, &stmt);
@@ -26161,7 +26125,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(id, mCursor->mIndexId);
+  rv = stmt->BindInt64ByName(kStmtParamNameId, mCursor->mIndexId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -26220,8 +26184,6 @@
                         : NS_LITERAL_CSTRING("index_data");
 
   NS_NAMED_LITERAL_CSTRING(sortColumn, "sort_column");
-  NS_NAMED_LITERAL_CSTRING(id, "id");
-  NS_NAMED_LITERAL_CSTRING(openLimit, " LIMIT ");
 
   nsAutoCString sortColumnAlias;
   if (mCursor->IsLocaleAware()) {
@@ -26239,7 +26201,7 @@
                                  "object_data_key "
                                  " FROM ") +
                              table + NS_LITERAL_CSTRING(" WHERE index_id = :") +
-                             id;
+                             kStmtParamNameId;
 
   nsAutoCString keyRangeClause;
   if (usingKeyRange) {
@@ -26270,7 +26232,7 @@
   // Note: Changing the number or order of SELECT columns in the query will
   // require changes to CursorOpBase::PopulateResponseFromStatement.
   nsCString firstQuery = queryStart + keyRangeClause + directionClause +
-                         openLimit + NS_LITERAL_CSTRING("1");
+                         kOpenLimit + NS_LITERAL_CSTRING("1");
 
   DatabaseConnection::CachedStatement stmt;
   nsresult rv = aConnection->GetCachedStatement(firstQuery, &stmt);
@@ -26278,7 +26240,7 @@
     return rv;
   }
 
-  rv = stmt->BindInt64ByName(id, mCursor->mIndexId);
+  rv = stmt->BindInt64ByName(kStmtParamNameId, mCursor->mIndexId);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
@@ -26457,10 +26419,6 @@
 
   nsCString query = continueQuery + countString;
 
-  NS_NAMED_LITERAL_CSTRING(currentKeyName, "current_key");
-  NS_NAMED_LITERAL_CSTRING(rangeKeyName, "range_key");
-  NS_NAMED_LITERAL_CSTRING(objectKeyName, "object_key");
-
   const bool usingRangeKey = !mCursor->mRangeKey.IsUnset();
 
   DatabaseConnection::CachedStatement stmt;
@@ -26471,20 +26429,20 @@
 
   int64_t id = isIndex ? mCursor->mIndexId : mCursor->mObjectStoreId;
 
-  rv = stmt->BindInt64ByName(NS_LITERAL_CSTRING("id"), id);
+  rv = stmt->BindInt64ByName(kStmtParamNameId, id);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   // Bind current key.
-  rv = currentKey.BindToStatement(stmt, currentKeyName);
+  rv = currentKey.BindToStatement(stmt, kStmtParamNameCurrentKey);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }
 
   // Bind range key if it is specified.
   if (usingRangeKey) {
-    rv = mCursor->mRangeKey.BindToStatement(stmt, rangeKeyName);
+    rv = mCursor->mRangeKey.BindToStatement(stmt, kStmtParamNameRangeKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -26495,7 +26453,7 @@
   if (isIndex && !hasContinueKey &&
       (mCursor->mDirection == IDBCursor::NEXT ||
        mCursor->mDirection == IDBCursor::PREV)) {
-    rv = mCursor->mObjectKey.BindToStatement(stmt, objectKeyName);
+    rv = mCursor->mObjectKey.BindToStatement(stmt, kStmtParamNameObjectKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
@@ -26504,7 +26462,7 @@
   // Bind object key if primaryKey is specified.
   if (hasContinuePrimaryKey) {
     rv = mParams.get_ContinuePrimaryKeyParams().primaryKey().BindToStatement(
-        stmt, objectKeyName);
+        stmt, kStmtParamNameObjectKey);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       return rv;
     }
